pkg/jobqueue/jobqueue.go

```diff
@@ -68,6 +72,50 @@ type JobQueue interface {
 	//
 	// Lastly, the IDs of the jobs dependencies are returned.
 	JobStatus(id uuid.UUID) (jobType string, channel string, result json.RawMessage, queued, started, finished time.Time, canceled bool, deps []uuid.UUID, dependents []uuid.UUID, err error)
+	// Does the same as JobStatus but without fetching the result blob
+	// To be used instead of JobStatus if possible
+	JobStatusWoResult(id uuid.UUID) (jobType string, channel string, queued, started, finished time.Time, canceled bool, deps []uuid.UUID, dependents []uuid.UUID, err error)
+
+	// Query multiple fields under the result column at once and stores them in to the response object.
+	// paths: a list of dot separated fields composing a searchable path in the result column
+	//        each names in the path must correspond to a subsequent json key.
+	// response: an object that has for characteristic to contains public fields associated with json tags matching the paths
+	//
+	// for example:
+	// type Object struct {
+	//     Name string `json:"name"`
+	// }
+	// type ResponseStruct struct {
+	//     Name  string `json:"name"`
+	//     Obj1  Object `json:"obj"`
+	//     Obj2 *Object `json:"obj2"`
+	// }
+	// var response ResponseStruct
+	// err = QueryResultFields(id, []string{"name", "obj.name", "obj2.name"), &response)
+	//
+	// After calling the QueryResultFields, response will contain its Name , Obj1 and Obj2 fields field up.
+	// Obj2 will be instantiated on the fly.
+	//
+	// Pointer instantiation only works for one level of indirection, for pointers on pointers the user needs to provide
+	// a valid chain to dereference.
+	//
+	// For instance:
+	// type Object struct {
+	//     Name string `json:"name"`
+	// }
+	// type ResponseStruct struct {
+	//     Obj **Object `json:"obj"`
+	// }
+	// obj := &Object{}
+	// var response ResponseStruct
+	// response.Obj = &obj
+	// err = QueryResultFields(id, []string{"name", "obj.name", "obj2.name"), &response)
+	QueryResultFields(id uuid.UUID, paths []string, response any) error
+
+	// Check for presence of a specific path and is content in a JSON field.
+	// path -> a dot separated path within the result, for example osbuild_output.error
+	// Returns a boolean set to True if the requested path contains data
+	TestResultFieldNotEmpty(id uuid.UUID, path string) (bool, error)
```

Comment:
The `Test` prefix is usually used for tests in Go, so I find this name slightly confusing. What about `IsResultFieldNotEmpty`?